src/accelerate/utils/launch.py

```diff
@@ -320,6 +320,17 @@ def prepare_deepspeed_cmd_env(args: argparse.Namespace) -> Tuple[List[str], Dict
         # set to default pdsh
         args.deepspeed_multinode_launcher = DEEPSPEED_MULTINODE_LAUNCHERS[0]
 
+    if args.deepspeed_multinode_launcher in DEEPSPEED_MULTINODE_MPI_LAUNCHERS:
+        # MPI based launchers do not support --include and --exclude
+        if args.deepspeed_exclusion_filter is not None:
+            raise ValueError(
+                f"--deepspeed_exclusion_filter is not supported with --deepspeed_multinode_launcher {args.deepspeed_multinode_launcher}"
+            )
+        if args.deepspeed_inclusion_filter is not None:
+            raise ValueError(
+                f"--deepspeed_inclusion_filter is not supported with --deepspeed_multinode_launcher {args.deepspeed_multinode_launcher}"
+            )
+
     if num_machines > 1 and args.deepspeed_multinode_launcher != DEEPSPEED_MULTINODE_LAUNCHERS[1]:
         cmd = ["deepspeed", "--no_local_rank"]
         cmd.extend(["--hostfile", str(args.deepspeed_hostfile), "--launcher", str(args.deepspeed_multinode_launcher)])
```

Comment:
regarding line 344, although out of scope of this PR, I don't think `--exclude` and `--include` need to be mutually exclusive